src/debuginfo.cpp

```diff
@@ -774,9 +775,11 @@ static int lookup_pointer(DIContext *context, jl_frame_t **frames,
 
     int fromC = (*frames)[0].fromC;
     int n_frames = inlineInfo.getNumberOfFrames();
-    if (n_frames == 0)
+    if (n_frames == 0) {
+        JL_UNLOCK(&codegen_lock);
```

Comment:
No, it'll dead lock if

1. thread 1 is holding the codegen lock in codegen
2. thread 2 trying to grab the lock with `_nogc`
3. a GC triggers in 1 (or on another thread and thread 1 hits a safepoint while the lock is still held)

Now the GC will wait for thread 2 forever since it cannot get the codegen lock and is waiting without checking safepoint.